### ISSUE TITLE
nearest_point runtime error metpy

### ISSUE THREAD

**Issue by DrakonianMight** [blaylockbk/Herbie#179]
Hi,

I seem to be having an issued with ds.nearest_points, with the following error appearing.
`---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
Cell In[190], line 1
----> 1 dsi = ds.herbie.nearest_points(
      2     points=[(-100, 40), (-115, 34)],
      3     names=["point_a", "point_b"],
      4 )

File ~/miniconda3/envs/mlenv/lib/python3.8/site-packages/herbie/accessors.py:264, in HerbieAccessor.nearest_points(self, points, names, verbose)
    260     ds = ds.metpy.parse_cf()
    262 # Apply the MetPy method `assign_y_x` to the dataset
    263 # https://unidata.github.io/MetPy/latest/api/generated/metpy.xarray.html?highlight=assign_y_x#metpy.xarray.MetPyDataArrayAccessor.assign_y_x
--> 264 ds = ds.metpy.assign_y_x()
    266 # Convert the requested [(lon,lat), (lon,lat)] points to map projection.
    267 # Accept a list of point tuples, or Shapely Points object.
    268 # We want to index the dataset at a single point.
    269 # We can do this by transforming a lat/lon point to the grid location
    270 crs = ds.metpy_crs.item().to_cartopy()

File ~/miniconda3/envs/mlenv/lib/python3.8/site-packages/metpy/xarray.py:1033, in MetPyDatasetAccessor.assign_y_x(self, force, tolerance)
   1030             grid_prototype = data_var
   1031         if (not force and (hasattr(data_var.metpy, 'y')
   1032                            or hasattr(data_var.metpy, 'x'))):
-> 1033             raise RuntimeError('y/x coordinate(s) are present. If you wish to '
   1034                                'overwrite these, specify force=True.')
   1036 # Calculate y and x from grid_prototype, if it exists, and assign
   1037 if grid_prototype is None:

RuntimeError: y/x coordinate(s) are present. If you wish to overwrite these, specify force=True.`
This appears to be coming from the MetPy.assign_X_y function and i don't appear to have an easy way of parsing a specify=force into the function.

Any ideas what is causing the issue?
I have tested on several different models and I get the same error.

Running python 3.8 conda forge installation

**Comment by blaylockbk** [comment]
Can you provide more details on what your Herbie object is?

I can't reproduce with 
- Python 3.11.3
- Metpy 1.4.1
- xarray 2023.2.0
- Herbie 2023.3.0

This worked

```python
from herbie import Herbie
H = Herbie("2023-01-01", model="hrrr")
ds = H.xarray("TMP:2 m")
dsi = ds.herbie.nearest_points(
    points=[(-100, 40), (-115, 34)],
    names=["point_a", "point_b"],
)
dsi
```
![image](https://user-images.githubusercontent.com/6249613/231506337-08b8fa73-05f4-4730-b5d4-235fe8c5d70c.png)

**Comment by DrakonianMight** [comment]
Hey, 

Versions are:
python                    3.8.16  
herbie-data               2023.3.0 
xarray                    2022.11.0 
metpy                     1.4.1 

The model type doesn't seem to matter, this error occurs with several models.
Here is an example with a GEFS grib file.

![image](https://user-images.githubusercontent.com/46990198/231590069-d3ed634f-a1b7-48e7-bd54-bb6ee871c25b.png)

**Comment by DrakonianMight** [comment]
Update, I have tried with a fresh environment on my M2 mac and I still get the same problem.
python=3.11
xarray=2023.2.0
herbie-data=2023.3.0
metpy=1.4.1

I still get the following error running this code:
`H = Herbie("2022-04-12 00:00", fxx=3, model="ecmwf", product="wave")
ds = H.xarray(":swh:sfc:g:0001:od:fc:wave", remove_grib=False)
ds
`
followed by
![image](https://user-images.githubusercontent.com/46990198/232181088-9f67b021-d71f-46f4-a585-e2bb2c20cb82.png)

**Comment by joshuaeh** [comment]
Not sure the cause, root cause, but the issue is thrown as metpy tries to assign latitude and longitude values to x and y coordinates but the dataset does not have x and y coordinates. This could be fixed by checking for the coordinates internally. I got around this by interpolating the xarray dataset directly: `dsi = ds.interp(latitude=my_latitude, longitude=my_longitude)`

**Comment by blaylockbk** [comment]
I believe I implemented the "picking values from points" accessor better in the new `ds.herbie.pick_points` accessor released in version 2024.5.0. Check it out here https://herbie.readthedocs.io/en/latest/user_guide/tutorial/accessor_notebooks/pick_points.html

Let me know if you have any issues.
